src/sql/query.rs

```diff
@@ -1836,6 +1836,52 @@ fn plan_function<'a>(
                 Ok(expr)
             }
 
+            "round" => {
+                // https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_round
+                // The return value has the same type as the first argument
+                // (assuming that it is integer, double, or decimal).
+                // This means that for an integer argument, the result is an integer (no decimal places)
+                if sql_func.args.is_empty() || sql_func.args.len() > 2 {
+                    bail!(
+                        "round expects 1 or 2 arguments, got {}",
+                        sql_func.args.len()
+                    );
+                }
+
+                let expr1 = plan_expr(ecx, &sql_func.args[0], None)?;
+                let typ1 = ecx.column_type(&expr1);
+
+                if sql_func.args.len() == 1 {
+                    let expr = match typ1.scalar_type {
+                        ScalarType::Int32 => expr1.call_unary(UnaryFunc::RoundInt32),
+                        ScalarType::Int64 => expr1.call_unary(UnaryFunc::RoundInt64),
+                        ScalarType::Float32 => expr1.call_unary(UnaryFunc::RoundFloat32),
+                        ScalarType::Float64 => expr1.call_unary(UnaryFunc::RoundFloat64),
+                        ScalarType::Decimal(_, s) => expr1.call_unary(UnaryFunc::RoundDecimal(s)),
+                        _ => bail!("round({}) does not exist", typ1.scalar_type),
+                    };
+                    Ok(expr)
+                } else {
+                    let expr2 = plan_expr(ecx, &sql_func.args[1], None)?;
+                    let expr2 = promote_int_int64(ecx, "round", expr2)?;
+                    let typ2 = ecx.column_type(&expr2);
+
+                    let expr = match typ1.scalar_type {
+                        ScalarType::Int32 => expr1.call_binary(expr2, BinaryFunc::RoundInt32),
+                        ScalarType::Int64 => expr1.call_binary(expr2, BinaryFunc::RoundInt64),
+                        ScalarType::Decimal(_, s) => {
```

Comment:
And for pgcompat the two argument form only needs an implementation for decimals, though we'll need a new function to promote non-floats to decimals.